### PR TITLE
[tracking] Mirage3 support

### PR DESCRIPTION
**Pull request by avsm**
Do not merge just yet, as it depends on an unreleased version of Mirage.

Can build it using the OPAM remote at https://github.com/mirage/mirage-dev.git

### CODE REVIEW

File: config.ml [hunk 1]
```diff
@@ -123,9 +123,9 @@ let packages = [
 (* Network stack *)
 
 let stack =
-  match get_mode () with
-  | `Xen -> generic_stackv4 default_console tap0
-  | `Unix | `MacOSX -> socket_stackv4 default_console [Ipaddr.V4.any]
+  if_impl Key.is_xen
```
Comment (hannesm) [inline config.ml]: more reliable: `is_unix` (since `is_xen` should be deprecated due to solo5 being around as well)

Comment (avsm) [reply config.ml]: fixed in 9e24e5be3171f5233101011fbc988cfaeb06cd2a

**Comment by hannesm** [comment]
I do like 37adc8b b1b2359 b8bc009 (maybe 07d5061), which should be a separate PR.

I did some more cosmetic changes at https://github.com/hannesm/Canopy/tree/new
 - https://github.com/hannesm/Canopy/commit/b54eeba6c40948f0bde611dfafc33bf12a753247 (enable markdown in abstract)
 - https://github.com/hannesm/Canopy/commit/a0d53987e5ae7db769cd06253e58807b3dd1574d (remove JS/bootstrap.css, unify, emit fewer HTML elements)

I also did a mirage3 port at https://github.com/hannesm/Canopy/commit/a781de10513e0ab59b53b8ba81ac0487e1e4f2c8

**Comment by amirmc** [comment]
Fixed in https://github.com/Engil/Canopy/pull/79

**Pull request closed without merging**